src/Core/AndroidNative/maui/src/main/java/com/microsoft/maui/PlatformInterop.java

```diff
@@ -66,6 +69,67 @@
 
 public class PlatformInterop {
 
+    /**
+     * Checks if the given context is destroyed.
+     * This is similar to the C# IsDestroyed extension method in ContextExtensions.cs
+     * @param context The context to check
+     * @return true if the context is destroyed, false otherwise
+     */
+    private static boolean isContextDestroyed(Context context) {
+        if (context == null) {
+            return true;
+        }
+
+        Activity activity = getActivity(context);
+        if (activity instanceof FragmentActivity) {
+            FragmentActivity fragmentActivity = (FragmentActivity) activity;
+            
+            // Check if activity is finishing or destroyed
+            if (fragmentActivity.isFinishing() || fragmentActivity.isDestroyed()) {
+                return true;
+            }
+
+            // Check lifecycle state
+            try {
+                if (fragmentActivity.getLifecycle().getCurrentState() == Lifecycle.State.DESTROYED) {
+                    return true;
+                }
+            } catch (Exception e) {
+                // If there's an exception getting the lifecycle state, consider it destroyed
```

Comment:
Should we log some warning here? It just completely swallows the error and you'd never see it?